samcli/commands/local/lib/local_function_url_service.py

```diff
@@ -0,0 +1,351 @@
+                "Add FunctionUrlConfig to your Lambda functions to use this feature.\\n"
+                "Example:\\n"
+                "  MyFunction:\\n"
+                "    Type: AWS::Serverless::Function\\n"
+                "    Properties:\\n"
+                "      FunctionUrlConfig:\\n"
+                "        AuthType: NONE"
+            )
+
+    def _allocate_port(self) -> int:
+        """
+        Allocate next available port in range
+
+        Returns
+        -------
+        int
+            An available port number
+
+        Raises
+        ------
+        PortExhaustedException
+            When no ports are available in the specified range
+        """
+        for port in range(self._port_start, self._port_end + 1):
+            if port not in self._used_ports:
+                # Actually check if the port is available by trying to bind to it
+                if self._is_port_available(port):
+                    self._used_ports.add(port)
+                    return port
+        raise PortExhaustedException(f"No available ports in range {self._port_start}-{self._port_end}")
+
+    def _is_port_available(self, port: int) -> bool:
+        """
+        Check if a port is available by attempting to bind to it
+
+        Parameters
+        ----------
+        port : int
+            Port number to check
+
+        Returns
+        -------
+        bool
+            True if port is available, False otherwise
+        """
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                sock.bind((self.host, port))
+                return True
+        except OSError:
+            LOG.debug(f"Port {port} is already in use")
+            return False
+
+    def _start_function_service(self, func_name: str, func_config: Dict, port: int) -> FunctionUrlHandler:
+        """Start individual function URL service"""
+        service = FunctionUrlHandler(
+            function_name=func_name,
+            function_config=func_config,
+            local_lambda_runner=self.invoke_context.local_lambda_runner,
+            port=port,
+            host=self.host,
+            disable_authorizer=self.disable_authorizer,
+            stderr=self.invoke_context.stderr,
+            ssl_context=None,
+        )
+        return service
+
+    def start(self):
+        """
+        Start the Function URL services. This method will block until stopped.
+        """
+        if not self.function_urls:
+            raise NoFunctionUrlsDefined("No Function URLs found to start")
+
+        # Setup signal handlers
+        def signal_handler(sig, frame):
+            LOG.info("Received interrupt signal. Shutting down...")
+            self._shutdown_event.set()
+
+        signal.signal(signal.SIGINT, signal_handler)
+        signal.signal(signal.SIGTERM, signal_handler)
```

Comment:
I see that these signals are handled on `BaseLocalService`. In the furl_handler file you're extending that class, but I think not really using it. If you use it correctly, then these signals should be handled there already so we shouldn't have to handle them again.